singleflight/example_test.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2016 The Go Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package singleflight_test
+
+import (
+	"fmt"
+	"log"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/groupcache/singleflight"
+)
+
+var counter int32
+var networkGroup singleflight.Group
+
+// expensiveNetworkCall simulates an expensive action.
+func expensiveNetworkCall() (interface{}, error) {
+	time.Sleep(1)
+	atomic.AddInt32(&counter, 1)
+	return counter, nil
```

Comment:
Yikes. Should be fixed now. Thanks for sticking with this through several rounds.